TOML Throws

trisyoungs opened the pull request:
Just a small PR to capture a minor modification I made while testing the TOML and forgot about!

trisyoungs committed:
Throw on unrecognised / missing GeometryType.

trisyoungs committed:
For Windows' benefit.

rprospero commented on src/base/geometry.h:
```diff
@@ -52,8 +53,11 @@ template <> struct from<Geometry::GeometryType>
             return Geometry::GeometryType::TorsionType;
+        else
+            throw(std::runtime_error(
+                fmt::format("Unhandled geometry type '{}' - can't convert from TOML value.\n", std::string(typeString))));
```
```suggestion
            throw toml::syntax_error(
                fmt::format("Unhandled geometry type '{}' - can't convert from TOML value.\n", std::string(typeString)), node.location());
```

Using the `toml::syntax_error` exception means that the error message will include the line number where the problem occurs.  Makes life much easier for the user debugging the input file.

trisyoungs replied:
This is something I now realise from working on the (de)serialise for my new keyword type!

rprospero commented on src/base/geometry.h:
```diff
@@ -69,6 +73,8 @@ template <> struct into<Geometry::GeometryType>
             case Geometry::GeometryType::TorsionType:
                 return "torsion";
+            default:
+                throw(std::runtime_error("Unhandled geometry type - can't convert to TOML value.\n"));
```
```suggestion
                throw toml::syntax_error("Unhandled geometry type - can't convert to TOML value.\n", node.location());
```

Same

trisyoungs committed:
Apply suggestions from code review

Co-authored-by: Adam Washington <adam.washington@stfc.ac.uk>

trisyoungs committed:
Formatting.

rprospero committed:
Fix exception for serialising geometry

You can't throw a line number for an error while *serialising*, since
the file doesn't exist yet.

As a more important note, this means that we need to wrap
serialisation in a catch statement, as well as deserialisation.

rprospero committed:
Fix whitespace
